hasjob/views/listing.py

```diff
@@ -560,49 +560,65 @@ def pinnedjob(domain, hashid):
-        post.review_datetime = datetime.utcnow()
-        post.reviewer = g.user
-
-        if request.form.get('submit') == 'spam':
+        if request.form.get('submit') == 'reject':
+            flashmsg = "This job post has been rejected."
+            post.status = POSTSTATUS.REJECTED
+            post.closed_datetime = db.func.utcnow()
+            post.review_comments = rejectform.reason.data
+            post.review_datetime = db.func.utcnow()
+            post.reviewer = g.user
+
+            send_reject_mail(request, 'reject', post)
+        elif request.form.get('submit') == 'spam':
             flashmsg = "This job post has been marked as spam."
             post.status = POSTSTATUS.SPAM
-        else:
-            if request.form.get('submit') == 'ban':
-                flashmsg = "This job post has been rejected and the user and domain banned."
-                post.domain.is_banned = True
-                post.domain.banned_by = g.user
-                post.domain.banned_reason = rejectform.reason.data
-                if post.user:
-                    post.user.blocked = True
-
-                # In case of Ban, Reject all the job posts by the same domain
-                # unless it's a webmail domain
-                if not post.domain.is_webmail:
-                    for jobpost in post.domain.jobposts:
-                        if jobpost.is_listed():
-                            jobpost.status = POSTSTATUS.REJECTED
-                            jobpost.closed_datetime = datetime.utcnow()
-                            jobpost.review_comments = rejectform.reason.data
-                            jobpost.review_datetime = datetime.utcnow()
-                            jobpost.reviewer = g.user
-            else:
-                flashmsg = "This job post has been rejected."
-                post.status = POSTSTATUS.REJECTED
-
-            msg = Message(subject="About your job post on Hasjob",
-                recipients=[post.email])
-            msg.html = email_transform(render_template('reject_email.html', post=post), base_url=request.url_root)
-            msg.body = html2text(msg.html)
-            mail.send(msg)
+            post.closed_datetime = db.func.utcnow()
+            post.review_comments = rejectform.reason.data
+            post.review_datetime = db.func.utcnow()
+            post.reviewer = g.user
+        elif request.form.get('submit') == 'ban':
+            flashmsg = "This job post has been rejected and the user and domain banned."
+            post.domain.is_banned = True
+            post.domain.banned_by = g.user
+            post.domain.banned_reason = rejectform.reason.data
+            if post.user:
+                post.user.blocked = True
+
+            # In case of Ban, Reject all the job posts by the same domain
+            # unless it's a webmail domain
+            if not post.domain.is_webmail:
+                for jobpost in post.domain.jobposts.options(*JobPost._defercols):
```

Comment:
@jace can I do this with back-references? Didn't test this part. Need to add some job posts with custom domain.